### PR TITLE
language.md complete revamp

### PR DESCRIPTION
**Pull request by virtualknight8**
## (ﾉ◕ヮ◕)ﾉ*✲ﾟ*｡⋆ You've made a PR!

Before the code PR is merged, please ensure that the following checklist is completed. 

- [ ] Branch name details the work done for the PR
- [ ] CI tests have passed
- [ ] An admin has reviewed and accepted the PR
- [ ] Clang format has been run on changes (`run ./tools/format.sh`)

Please be sure that the code follows the style as follows:

- Everything in `snake_case`
- Classes end in `_c` (unless its an interface, then end in `_if`)
- Structs end in `_s`
- Unions end in `_u`
- Enumerations end in `_e`
- Type names that are redefined via `using` end in `_t` (always use `using` - not `typedef`)
- Type names that define a particular type of pointer end in `_ptr`

The above is just a loose formatting that has been used already throughout the C++ source code of Nibi,
and we would like to keep things mostly uniform. In the future, an actual style guide may be created. 

## Description of work:

Please enter a description of your work here:
```
verified and updated the corresponding file to be more comprehensive and error free.

```

### CODE REVIEW

File: docs/LANGUAGE.md [hunk 1]
```diff
@@ -1,322 +1,295 @@
...
-| exit    | Exit the program | N/A
-| eval    | Evaluate a given string | variable
-| quote   | Quote a a list into a string | variable
-| nop     | Do nothing | nil
-| macro   | Define a macro | variable
-| dict | Create a dictionary | the new dictionary
-| extern-call | Call a c-function from a shared library | variable
-| alias   | Use a new symbol to refer to the data behing another | nil
-| exchange | Update the value of a cell and return the old value of the cell | variable
-| str-set-at | Update a string by inserting a value at a given index (negative indexing permitted) | updated string
```
Comment (bosley) [inline docs/LANGUAGE.md:69]: These do not appear in the new changes. There were a couple of additions while the changes you have were being made. I'll hop around and point them out.

File: docs/LANGUAGE.md [hunk 2]
```diff
@@ -1,322 +1,295 @@
...
-| split   | Convert an item to a list comprised of the raw elements of the given variable | converted value 
-| str-lit | Convert a list into a direct string rather than just a string representation | new string
-
-| memory commands | description | returns
-|----   |---- |----
-| mem-new | Allocate some memory manually on the heap | Pointer cell|
-| mem-del | Delete some memory on the heap | Pointer cell |
-| mem-cpy | Copy a trivial cell's data to heap, or copy a pointer cell | Destination pointer cell |
-| mem-load | Load a data from heap into a new cell | New cell of trivial cell type |
-| mem-is-set | Check if a cell pointer has its pointer set to a space in memory | T/F |
```
Comment (bosley) [inline docs/LANGUAGE.md:97]: These are the current manual memory management instructions; fewer than that showing in the changes. A couple of the instructions included in the MR no longer exist.

File: docs/LANGUAGE.md [hunk 3]
```diff
@@ -1,322 +1,295 @@
...
+| `nop`           | Do nothing | `nil` |
+| `macro`         | Define a macro | Variable |
+| `dict`          | Create a dictionary | The new dictionary |
+| `extern-call`   | Call a C-function from a shared library | Variable |
+| `alias`         | Use a new symbol to refer to the data behind another | `nil` |
+
+| Type Commands | Description | Returns |
+| ------------- | ----------- | ------- |
+| `type`        | Retrieve a string detailing the type of a given item | String |
+| `str`         | Convert an item to a string type | Converted value |
```
Comment (bosley) [inline docs/LANGUAGE.md:61]: missing `char` (another recent addition)

File: docs/LANGUAGE.md [hunk 4]
```diff
@@ -460,252 +427,262 @@ Keyword: `assert`
...
+| ptr             | A pointer                                            |
+| string          | A string cell                                        |
+| list:data       | A data list                                          |
+| list:access     | An access list                                       |
+| list:instruction | An instruction list                                  |
+| list            | Undetermined list type                               |
+| function        | Function                                             |
+| environment     | An environment of cells                              |
+| symbol          | A symbol                                             |
+| dict            | A dictionary                                         |
```
Comment (bosley) [inline docs/LANGUAGE.md:563]: It wasn't added in there before, but `char` could also be returned as a type. its a simple c char

File: docs/LANGUAGE.md [hunk 5]
```diff
@@ -840,239 +821,171 @@ Example:
...
+- Keyword: `mem-acquire`
+  - Marks a given ptr cell as being owned by the Nibi runtime.
+  - By default, if a pointer cell is made via `mem-new`, it is considered owned. If the pointer comes from an external source, it is not owned.
+  - Acquiring ownership of the pointer should only be done if the external source expects the caller to manage the resource's lifetime.
+  - Deleting a manually acquired pointer value may cause UB.
+
+- Keyword: `mem-abandon`
+  - Marks any N number of pointers as no longer owned by the runtime.
+  - Runtime will not be able to delete the pointer unless it is later re-acquired.
+  - This is meant to be done after moving ownership of data to an external source and indicating to the runtime that the pointer is now externally managed.
```
Comment (bosley) [inline docs/LANGUAGE.md:977]: These were removed

**Review by bosley — CHANGES_REQUESTED** [review]
Looks really good, but some changes are required to include information added after task started.

File: docs/LANGUAGE.md (hunk 1 above)
Comment (virtualknight8) [reply docs/LANGUAGE.md:69]: Just to be clear, I'm assuming I have to add them because they got removed.

File: docs/LANGUAGE.md (hunk 2 above)
Comment (virtualknight8) [reply docs/LANGUAGE.md:97]: So, only these 5 need to exist, right ?

File: docs/LANGUAGE.md (hunk 3 above)
Comment (virtualknight8) [reply docs/LANGUAGE.md:61]: Oh ok. I think I got this one.

File: docs/LANGUAGE.md (hunk 4 above)
Comment (virtualknight8) [reply docs/LANGUAGE.md:563]: I think I understand this as well.

File: docs/LANGUAGE.md (hunk 5 above)
Comment (virtualknight8) [reply docs/LANGUAGE.md:977]: And these need to go ?

File: docs/LANGUAGE.md (hunk 2 above)
Comment (bosley) [reply docs/LANGUAGE.md:97]: Yeah, if you look at what the file diff here, the spot in red is what you've changed. They only ones that existed to be updated at the time of the PR were:

```
| mem-new | Allocate some memory manually on the heap | Pointer cell|
| mem-del | Delete some memory on the heap | Pointer cell |
| mem-cpy | Copy a trivial cell's data to heap, or copy a pointer cell | Destination pointer cell |
| mem-load | Load a data from heap into a new cell | New cell of trivial cell type |
| mem-is-set | Check if a cell pointer has its pointer set to a space in memory | T/F |
```

**Review by bosley — APPROVED** [review]

**Pull request merged**